relay-server/src/actors/envelopes.rs

```diff
@@ -449,199 +447,53 @@ impl Handler<QueueEnvelope> for EnvelopeManager {
     }
 }
 
-/// Handles a queued envelope.
-///
-/// 1. Ensures the project state is up-to-date and then validates the envelope against the state and
-///    cached rate limits. See [`CheckEnvelope`] for full information.
-/// 2. Executes dynamic sampling using the sampling project.
-/// 3. Runs the envelope through the [`EnvelopeProcessor`] worker pool, which parses items, applies
-///    normalization, and runs filtering logic.
-/// 4. Sends the envelope to the upstream or stores it in Kafka, depending on the
-///    [`processing`](Config::processing_enabled) flag.
-/// 5. Captures [`Outcome`]s for dropped items and envelopes.
-///
-/// This operation is invoked by [`QueueEnvelope`] for envelopes containing all items except
-/// metrics.
-struct HandleEnvelope {
+/// Sends an envelope to the upstream or Kafka.
+pub struct SubmitEnvelope {
```

Comment:
Note that `SubmitEnvelope` is not a replacement for `HandleEnvelope`. Instead, it implements just a subset of what had been there before, namely the sending part.